scipy/optimize/_quasi_newton_approx.py

```diff
@@ -0,0 +1,261 @@
+"""quasi-Newton Hessian approximations"""
```

Comment:
I will rename the file to "_hessian_update". I think it is better than "_hessian_approximation" since finite-difference (and other types) of approximations are not implemented in this class.

The only weakness of this new name is that this class also implements the update of the inverse Hessian, so the name is not entirely accurate. Nevertheless this is a minor details and I think the name is indeed better.

Reply:
Actually after reading your next comment, I decided to rename the file to ``_hessian_update_strategy.py``, so the file and the class have the same name, facilitating the code navigation. What do you think?

Reply:
It's good.